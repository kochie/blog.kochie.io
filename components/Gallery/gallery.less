@import '../common.less';

.container {
  // margin-left: 10%;
  // margin-right: 10%;
  background-color: #f4f8fb;
  padding: 5%;
  display: grid;
  // width: 100%;
  grid-template-columns: repeat(1, 1fr);
  grid-template-rows: auto;
  grid-gap: 40px;

  @media (min-width: @MediumWidth) and (orientation: landscape) {
    grid-template-columns: repeat(4, 1fr);
    grid-auto-flow: row dense;
    padding-left: 10%;
    padding-right: 10%;
  }

  @media (min-width: @LargeWidth) and (orientation: landscape) {
    grid-template-columns: repeat(6, 1fr);
    padding-left: 10%;
    padding-right: 10%;
  }
}
