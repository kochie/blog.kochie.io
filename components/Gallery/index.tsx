import React, { useState, useEffect } from 'react'

import { Small, Medium, Large } from '../ArticleCards';

import style from "./gallery.less"
import { article } from 'articles.json';

const cardOrder = [Large, Small, Small, Small, Medium, Medium]

export default () => {
    const [cards, setCards] = useState<article[]>([])

    function calcCards(articles: article[]) {
        return articles.map((article, i) => {
            const Card = cardOrder[i%cardOrder.length]
            return (
                <Card 
                    title={article.title} 
                    tags={article.tags} 
                    image={article.jumbotron} 
                    articleName={article.articleFile} 
                    readTime={article.readTime} 
                    blurb={article.blurb} 
                />
            )
        })
    }

    async function getArticles() {
        const articles: article[] = await require("../../static/articles.json")
        setCards(articles)
    }

    useEffect(() => {
        getArticles()
    })

    return (
        <div className={style.container}>
            {calcCards(cards)}
        </div>
    )
}