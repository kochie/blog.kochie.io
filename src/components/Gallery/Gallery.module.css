.container {
  /* margin-left: 10%; */
  /* margin-right: 10%; */
  background-color: var(--gallery-background-color);
  padding: 25px;
  display: grid;
  grid-template-columns: repeat(6, 1fr);
  /* width: 100%; */
  /* grid-template-columns: repeat(1, 1fr); */
  grid-template-rows: auto;
  max-width: 1080px;
  margin: 0 auto 100px auto;
  grid-gap: 20px;
  grid-auto-flow: row dense;

  @media (--medium-viewport) and (orientation: landscape) {
    /* grid-template-columns: repeat(4, 1fr); */
    grid-gap: 30px;
    /* padding-left: 10%; */
    /* padding-right: 10%; */
  }

  @media (--large-viewport) and (orientation: landscape) {
    grid-gap: 40px;
    /* padding-left: 10%; */
    /* padding-right: 10%; */
  }
}
