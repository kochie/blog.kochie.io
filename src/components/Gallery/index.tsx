import React, { ReactElement } from 'react'
import { LargeCard, MediumCard, SmallCard } from '@/components/ArticleCards'
import type { CardDetails } from '@/components/ArticleCards'

import style from './Gallery.module.css'
import { ArticleMetadata } from '@/lib/article-path'

type CardElement = (CardDetails: CardDetails) => JSX.Element

interface GalleryProps {
  articles: ArticleMetadata[]
  cardOrder?: CardElement[]
  backgroundColor?: string
}

const Gallery = ({
  articles,
  cardOrder = [
    LargeCard,
    SmallCard,
    SmallCard,
    SmallCard,
    MediumCard,
    MediumCard,
  ],
  backgroundColor = 'transparent',
}: GalleryProps): ReactElement => {
  function calcCards(cards: ArticleMetadata[]): ReactElement[] {
    return cards.map((article, i): ReactElement => {
      const Card = cardOrder[i % cardOrder.length]
      return (
        <Card
          key={article.articleDir}
          title={article.title}
          tags={article.tags}
          image={article.jumbotron}
          articleDir={article.articleDir}
          readTime={article.readTime}
          blurb={article.blurb}
        />
      )
    })
  }

  return (
    <div className={style.container} style={{ backgroundColor }}>
      {calcCards(articles)}
    </div>
  )
}

export default Gallery
