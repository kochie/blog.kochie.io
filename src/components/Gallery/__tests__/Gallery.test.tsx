import React from 'react'
import { ReactTestRenderer, act, create } from 'react-test-renderer'
import Gallery from '@/components/Gallery'

describe('GALLERY COMPONENT', () => {
  test('renders correctly', () => {
    let tree: ReactTestRenderer

    act(() => {
      tree = create(<Gallery articles={[]} />)
    })

    // @ts-expect-error tree will be assigned
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
