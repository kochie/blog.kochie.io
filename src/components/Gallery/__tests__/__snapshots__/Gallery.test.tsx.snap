// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GALLERY COMPONENT renders correctly 1`] = `
<div
  className="container"
  style={
    {
      "backgroundColor": "transparent",
    }
  }
/>
`;
